var/spack/repos/builtin/packages/nf-tower-agent/package.py

```diff
@@ -8,7 +8,7 @@
 from spack.package import *
 
 
-class TowerAgent(Package):
+class NfTowerAgent(Package):
```

Comment:
I usually deprecate the old package when renaming instead of completely deleting it: https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions

Reply:
Very keen on your input on this one, Adam.

I had added `tower-agent` and `tower-cli` earlier this year. Then, while adding `nf-core-tools` with this PR, I have thought it would be friendlier for users to have the first two renamed. (I also believe so far the uptake will have still been low, in terms of spack installations for the former two).
So, according  to the doc page above, it is not even reason number 6 .. the tools themselves have not changed name, it is just a name change for the Spack recipe. And by the way, apologies for this change on the go, I will make sure it will not happen with future Spack contributions from me.

What are your thoughts?

Reply:
Number 6 "package rename" refers to the name of the Spack recipe. Or at least, that was my intention when I wrote those docs.

Since you're maintaining these recipes I will likely let you make the final decision. We do frown on package/version rename/removal without first deprecating it and waiting a full release (you can obviously create the new package at the same time as you deprecate the old package), but I don't think we have a hard policy about this.

Reply:
ok, thanks for clarifying Adam.
I have added back the old named packages, and added the `deprecated` property to all their versions